authorization/http_server.go

```diff
@@ -331,6 +330,39 @@ func newPermissionsResponse(ctx context.Context, ps []influxdb.Permission, svc i
 	return res, nil
 }
 
+func (h *AuthHandler) getNameForResource(ctx context.Context, resource influxdb.ResourceType, id influxdb.ID) (string, error) {
+	if err := resource.Valid(); err != nil {
+		return "", err
+	}
+
+	if ok := id.Valid(); !ok {
+		return "", influxdb.ErrInvalidID
+	}
+
+	switch resource {
+	case influxdb.BucketsResourceType:
+		r, err := h.tenantService.FindBucketByID(ctx, id)
+		if err != nil {
+			return "", err
+		}
+		return r.Name, nil
+	case influxdb.OrgsResourceType:
+		r, err := h.tenantService.FindOrganizationByID(ctx, id)
+		if err != nil {
+			return "", err
+		}
+		return r.Name, nil
+	case influxdb.UsersResourceType:
+		r, err := h.tenantService.FindUserByID(ctx, id)
+		if err != nil {
+			return "", err
+		}
+		return r.Name, nil
+	}
+
+	return "", nil
```

Comment:
Should we return an error here if something is attempting to map a resource type we don't support inside this structure?

Reply:
I think it should be okay since it is an optional field